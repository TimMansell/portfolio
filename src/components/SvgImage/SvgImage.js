import React from 'react';
import PropTypes from 'prop-types';

import styles from './SvgImage.module.scss';

export const SvgImage = ({ src, name, width }) => {
  return (
    <img
      src={src}
      alt={name}
      className={styles.img}
      width={width}
      title={name}
    />
  );
};

SvgImage.propTypes = {
  src: PropTypes.oneOfType([PropTypes.string, PropTypes.object]).isRequired,
  name: PropTypes.string.isRequired,
  width: PropTypes.string.isRequired,
};

export default SvgImage;
