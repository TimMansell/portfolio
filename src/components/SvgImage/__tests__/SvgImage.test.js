import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import SvgImage from '../SvgImage';

const props = {
  src: 'test src',
  name: 'test name',
  width: '20',
};

describe('SvgImage', () => {
  it('should render my component', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<SvgImage {...props} />);
  });

  it('should match snapshot', () => {
    const snapshot = renderer.create(<SvgImage {...props} />).toJSON();

    expect(snapshot).toMatchSnapshot();
  });

  it('should have correct img properties', () => {
    const wrapper = shallow(<SvgImage {...props} />);

    expect(wrapper.props().src).toBe(props.src);
    expect(wrapper.props().alt).toBe(props.name);
    expect(wrapper.props().width).toBe(props.width);
    expect(wrapper.props().title).toBe(props.name);
  });
});
