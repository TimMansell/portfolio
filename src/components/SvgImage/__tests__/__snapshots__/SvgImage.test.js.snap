// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgImage should match snapshot 1`] = `
<img
  alt="test name"
  className="img"
  loading="lazy"
  src="test src"
  title="test name"
  width="20"
/>
`;
